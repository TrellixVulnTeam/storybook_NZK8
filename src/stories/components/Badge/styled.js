// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

/**
 * Determine the badge's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.paleGrey;
    case 'green':
      return colors.shamrockGreen;
    case 'blue':
      return colors.iceBlue;
    case 'yellow':
      return colors.lightTan;
    default:
      return colors.peach;
  }
}

/**
 * Determine the badge's font color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getFontColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.white;
    case 'blue':
      return colors.dodgerBlue;
    case 'yellow':
      return colors.burntYellow;
    case 'green':
      return colors.white;
    default:
      return colors.coral;
  }
}

export const Wrapper = styled.div`
  display: ${p => (p.theme === 'none' ? 'none' : 'inline-block')};
  padding: 0 8px;
  height: 24px;
  border-radius: 4px;
  background-color: ${p => getAccentColor(p.theme)};
  color: ${p => getFontColor(p.theme)};
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.3px;
  text-align: center;
`;
