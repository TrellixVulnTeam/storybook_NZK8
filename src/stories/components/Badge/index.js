// @flow

import React from 'react';
import { Wrapper } from './styled';

type Props = {
  children: React$Node,
  theme?: 'none' | 'gray' | 'red' | 'blue' | 'yellow' | 'green',
};

export default (props: Props) => <Wrapper theme={props.theme}>{props.children}</Wrapper>;
