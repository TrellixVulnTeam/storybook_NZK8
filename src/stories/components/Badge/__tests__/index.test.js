import React from 'react';
import { mount } from 'enzyme';
import Badge from 'components/Badge/index';

describe('<Badge />', () => {
  it('should render correctly without theme', () => {
    const sut = mount(<Badge />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with none theme', () => {
    const sut = mount(<Badge theme="none" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with gray theme', () => {
    const sut = mount(<Badge theme="gray" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with red theme', () => {
    const sut = mount(<Badge theme="red" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with blue theme', () => {
    const sut = mount(<Badge theme="blue" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with yellow theme', () => {
    const sut = mount(<Badge theme="yellow" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with green theme', () => {
    const sut = mount(<Badge theme="green" />);

    expect(sut).toMatchSnapshot();
  });
});
