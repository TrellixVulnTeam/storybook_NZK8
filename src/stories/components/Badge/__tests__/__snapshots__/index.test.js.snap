// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render correctly with blue theme 1`] = `
<_default
  theme="blue"
>
  <styled__Wrapper
    theme="blue"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 FugZA"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly with gray theme 1`] = `
<_default
  theme="gray"
>
  <styled__Wrapper
    theme="gray"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 hXniGS"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly with green theme 1`] = `
<_default
  theme="green"
>
  <styled__Wrapper
    theme="green"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 jgFdPO"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly with none theme 1`] = `
<_default
  theme="none"
>
  <styled__Wrapper
    theme="none"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 inGZnF"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly with red theme 1`] = `
<_default
  theme="red"
>
  <styled__Wrapper
    theme="red"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 jVPPFe"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly with yellow theme 1`] = `
<_default
  theme="yellow"
>
  <styled__Wrapper
    theme="yellow"
  >
    <div
      className="styled__Wrapper-sc-10qy68h-0 nhnDB"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<Badge /> should render correctly without theme 1`] = `
<_default>
  <styled__Wrapper>
    <div
      className="styled__Wrapper-sc-10qy68h-0 jVPPFe"
    />
  </styled__Wrapper>
</_default>
`;
